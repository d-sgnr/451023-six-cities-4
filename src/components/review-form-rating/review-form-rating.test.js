import React from "react";
import renderer from "react-test-renderer";
import ReviewFormRating from "./review-form-rating.jsx";

it(`ReviewFormRating should be rendered correctly`, () => {
  const tree = renderer.create(
      <ReviewFormRating
        ratingNumber={1}
        onRatingChange={() => {}}
        isChecked={false}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`ReviewFormRating should be rendered correctly when checked`, () => {
  const tree = renderer.create(
      <ReviewFormRating
        ratingNumber={2}
        onRatingChange={() => {}}
        isChecked={true}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
