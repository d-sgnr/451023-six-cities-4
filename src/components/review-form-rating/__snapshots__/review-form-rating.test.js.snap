// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFormRating should be rendered correctly 1`] = `
Array [
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="1-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="1"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`ReviewFormRating should be rendered correctly when checked 1`] = `
Array [
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    id="2-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="2"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="2-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
